tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php

```diff
@@ -863,6 +863,25 @@ private function skipOci8AndMysqli()
             $this->markTestSkipped('Mysqli driver dont support this feature.');
         }
     }
+
+    public function testIteratorCanBeIteratedMultipleTimes()
+    {
+        $sql = "SELECT test_int, test_string FROM fetch_table";
+        $stmt = $this->_conn->query($sql);
+        $stmt->setFetchMode(\PDO::FETCH_ASSOC);
+
+        $i = 0;
+        foreach ($stmt as $row) {
+            $i++;
+        }
+
+        $j = 0;
+        foreach ($stmt as $row2) {
+            $j++;
+        }
+
+        $this->assertEquals($i, $j);
```

Comment:
Thanks for your reply and sorry for my delay. I've pushed something which I'm pretty sure is what you mean. However, I don't think this works. I can't seem to get microseconds. Only thing I can do to get around this is to add a sleep call 😞 . Any ideas?